common/djangoapps/enrollment/views.py

```diff
@@ -489,3 +490,16 @@ def post(self, request):
                     ).format(username=username, course_id=course_id)
                 }
             )
+        finally:
+            # Assumes that the ecommerce service uses an API key to authenticate.
+            if has_api_key_permissions:
+                current_enrollment = api.get_enrollment(username, unicode(course_id))
```

Comment:
ah, i see.   sorry, i misread the intent - we're looking at requested result vs. actual result, whereas I thought the previous state was somewhere in the mix.

Had I realized that I wouldn't have requested the `or None` stuff but it doesn't hurt to have it there IMO.